src/wayland/shell/xdg/decoration.rs

```diff
@@ -103,18 +104,30 @@ pub struct XdgDecorationState {
     global: GlobalId,
 }
 
+/// Data associated with a XdgDecorationManager global.
+#[allow(missing_debug_implementations)]
+pub struct XdgDecorationManagerGlobalData {
+    filter: Box<dyn for<'c> Fn(&'c Client) -> bool + Send + Sync>,
+}
+
 impl XdgDecorationState {
     /// Creates a new delegate type for handling xdg decoration events.
     ///
     /// A global id is also returned to allow destroying the global in the future.
-    pub fn new<D>(display: &DisplayHandle) -> XdgDecorationState
+    pub fn new<D, F>(display: &DisplayHandle, filter: F) -> XdgDecorationState
```

Comment:
imo it would be nice to keep the current path and extend `XdgDecorationState` with a `with_filter` constructor like other protocol abstractions already provide. In most cases there is not need to filter out this global and if I understood the linked issue correct you also "just" need this as a workaround for a borked client implementation.